tests/test_cv_optim.py

```diff
@@ -4,121 +4,120 @@
 from flexcode.regression_models import NN, RandomForest, XGBoost, CustomModel
 
 
-def test_coef_predict_same_as_predict_NN():
-    # Generate data p(z | x) = N(x, 1)
-  def generate_data(n_draws):
+BUMP_THRESHOLD_GRID = np.linspace(0, 0.2, 3)
+SHARPEN_GRID = np.linspace(0.5, 1.5, 3)
+
+def generate_data(n_draws):
+    """Generate data p(z | x) = N(x, 1)
+
+    Parameters
+    ----------
+    n_draws : int
+        number of samples to generate
+
+    Returns
+    -------
+    x : List[float]
+        Samples drawn from a 0, 1 normal distribution
+    z : List[float]
+        Samples drawn from a `x`, 1 normal distribution. Where `x` is random variate
+        created earlier.
+    """
     x = np.random.normal(0, 1, n_draws)
     z = np.random.normal(x, 1, n_draws)
     return x, z
 
-  x_train, z_train = generate_data(5000)
-  x_validation, z_validation = generate_data(5000)
-  x_test, z_test = generate_data(5000)
+def test_coef_predict_same_as_predict_NN():
+    x_train, z_train = generate_data(1000)
+    x_validation, z_validation = generate_data(1000)
+    x_test, z_test = generate_data(1000)
 
-  # Parameterize model
-  model = flexcode.FlexCodeModel(NN, max_basis=31, basis_system="cosine",
-                                 regression_params={"k": [5, 10, 20, 25, 30]})
+    # Parameterize model
+    model = flexcode.FlexCodeModel(NN, max_basis=31, basis_system="cosine",
+                                    regression_params={"k": [5, 10, 20, 25, 30]})
 
-  # Fit and tune model
-  model.fit(x_train, z_train)
-  model.tune(x_validation, z_validation,
-             bump_threshold_grid = np.linspace(0, 0.2, 3),
-             sharpen_grid = np.linspace(0.5, 1.5, 3))
+    # Fit and tune model
+    model.fit(x_train, z_train)
+    model.tune(x_validation, z_validation,
+                bump_threshold_grid = BUMP_THRESHOLD_GRID,
+                sharpen_grid = SHARPEN_GRID)
 
-  cdes_predict, z_grid = model.predict(x_test, n_grid=200)
+    cdes_predict, z_grid = model.predict(x_test, n_grid=200)
 
-  coefs = model.predict_coefs(x_test)
-  cdes_coefs = coefs.evaluate(z_grid)
+    coefs = model.predict_coefs(x_test)
+    cdes_coefs = coefs.evaluate(z_grid)
 
-  assert np.max(np.abs(cdes_predict - cdes_coefs)) <= 1e-4
+    assert np.max(np.abs(cdes_predict - cdes_coefs)) <= 1e-4
 
 
 def test_coef_predict_same_as_predict_RF():
-    # Generate data p(z | x) = N(x, 1)
-  def generate_data(n_draws):
-    x = np.random.normal(0, 1, n_draws)
-    z = np.random.normal(x, 1, n_draws)
-    return x, z
-
-  x_train, z_train = generate_data(5000)
-  x_validation, z_validation = generate_data(5000)
-  x_test, z_test = generate_data(5000)
+    x_train, z_train = generate_data(1000)
+    x_validation, z_validation = generate_data(1000)
+    x_test, _ = generate_data(1000)
```

Comment:
I've reduced almost all of these values from 5000 to 1000. The result is that the tests now run in about 15 seconds instead of ~40 seconds each. And there doesn't seem to be any impact to the results. Which, as near as I can tell, for these unit tests, should be independent of the training/validation/test data sizes.

Reply:
Could you reduce further by creating a single data set for use in all tests? e.g. a pytest fixture with package scope.

Reply:
Yes, I was considering that approach, and will likely incorporate that before this PR is fine. Fortunately, even when the requested size is 5 or 10k, these operations are very fast - <<1 second. 

But to your point, it would be better to include them as a pytest.fixture, and I'll plan on doing that in a follow up commit.

Reply:
Follow up to this. I was having a little trouble with the parameterization of the pytest fixture. So I went with just a plain method in the conftest.py file instead of a parameterized fixture that returns an object that would be unpacked.